scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -685,10 +685,8 @@ def apply_filters(self, **kwargs):
                 b = set(filter(lambda item: item.name in ts.platform_allow, self.platforms))
                 c = a.intersection(b)
                 if not c:
-                    _platform_scope = list(filter(lambda item: item.name in ts.platform_allow, \
+                    platform_scope = list(filter(lambda item: item.name in ts.platform_allow, \
```

Comment:
@PerMac, `platform_scope` might be set empty here, whereas current implementation keeps it in case filter returns empty list here.  Is it expected for this fix ?

Reply:
@golowanow I brought back the logic to how it was before. I also believe the check for `len(platform_scope)>0` is redundant:
- the block starts with `if ts.platform_allow`, hence platform_allow has to be non-empty. 
- `self.platforms` is also non-empty and contains all zephyr's platforms